docs/agent-api.asciidoc

```diff
@@ -315,6 +315,39 @@ Note that if a custom logger is provided, the `logLevel` option will be ignored.
 
 Whether or not the agent should monitor for uncaught exceptions and send them to the APM Server automatically.
 
+[[capture-location-stack-traces]]
+===== `captureLocationStackTraces`
+
+* *Type:* Number
+* *Default:* `1`
```

Comment:
Just an idea, but you have constants like `CAPTURE_LOCATION_STACK_TRACE_ALL` and `CAPTURE_LOCATION_STACK_TRACE_NON_ERRORS`. Instead of using a magic number here, perhaps we should encourage people to use those constants? 

For environment variables, using strings are an option too.

Reply:
Yeah, I was thinking of this, but came to the conclusion that since these wasn't available when setting environment variables, we had to reveal the numbers anyway, so we couldn't hide them behind constants.

Using strings in the environment variables is an interesting approach that I hadn't thought of. Right now though, we don't have a way in our config logic to handle the case where the environment variables should be translated from strings to numbers when [initially parsed](https://github.com/elastic/apm-agent-nodejs/blob/master/lib/config.js).

So it would make that logic more complicated, so I wonder if it's worth it?

Reply:
Good point. I'll change it to use strings instead

Reply:
I have string values for similar settings in the Python agent, e.g.: https://www.elastic.co/guide/en/apm/agent/python/2.x/configuration.html#config-collect-local-variables

Maybe makes sense to align?

Reply:
@beniwohli Good idea, though looking over it I don't see much we can re-use. I was thinking of going with `off`, `non-errors` and `all` (maybe `log` instead of `non-errors`... can't really decide).

Yours are for a slightly different use-case. Only overlap I see is `all` and potentially `none` instead of `off`, though I think `off` better conveys its meaning in this use-case.

Reply:
@roncohen would you still suggest that the users should use the constants on the config module or should they just type the strings manually? If it's the former, we have to expose them in a nicer way. Right now they have to do something like this:

```js
var apmConfig = require('elastic-apm-node/lib/config')
var apm = require('elastic-apm-node').start({
  captureLocationStackTraces: apmConfig.CAPTURE_LOCATION_STACK_TRACE_ALL
})
```

Reply:
Updated: No uses the possible values of `off`, `non-errors` and `all`. Question still remains if `non-errors` is a good value and if we should expose the config constants

Reply:
using the strings directly is probably fine - people will have to do that anyway if they use the env vars.  However, it would be great to check at configuration time/start-up that the values are set correctly to avoid spelling errors.